integration/cluster.go

```diff
@@ -687,17 +686,34 @@ func (m *member) Close() {
 		m.grpcServer.Stop()
 		m.grpcServer = nil
 	}
-	m.s.Stop()
+	if leaderTransfer {
+		m.s.Stop()
+	} else {
+		m.s.StopServer()
+	}
 	for _, hs := range m.hss {
 		hs.CloseClientConnections()
 		hs.Close()
 	}
 }
 
+// Close stops the member's etcdserver and closes its connections
+func (m *member) Close() {
+	m.closeServer(false)
+}
+
 // Stop stops the member, but the data dir of the member is preserved.
 func (m *member) Stop(t *testing.T) {
 	plog.Printf("stopping %s (%s)", m.Name, m.grpcAddr)
-	m.Close()
+	m.closeServer(false)
+	m.hss = nil
+	plog.Printf("stopped %s (%s)", m.Name, m.grpcAddr)
+}
+
+// StopWithAutoLeaderTransfer stops the member with auto leader transfer.
+func (m *member) StopWithAutoLeaderTransfer(t *testing.T) {
+	plog.Printf("stopping %s (%s)", m.Name, m.grpcAddr)
+	m.closeServer(true)
```

Comment:
@heyitsanthony Didn't do the `m.s.transferleader; m.s.stop` because `closeServer` also does extra stuff, like closing grpc server.